src/watcher/watcher.ts

```diff
@@ -40,28 +41,62 @@ export function createWatcher(opts: Opts) {
 
   const pluginWatchContributions = opts.plugins.reduce(
     (patterns, p) =>
-      patterns.concat(p.dev.addToSettings.watchFilePatterns || []),
+      patterns.concat(p.dev.addToWatcherSettings.watchFilePatterns ?? []),
     [] as string[]
   )
 
   const pluginIgnoreContributions = opts.plugins.reduce(
     (patterns, p) =>
-      patterns.concat(p.dev.addToSettings.ignoreFilePatterns || []),
+      patterns.concat(
+        p.dev.addToWatcherSettings.core?.ignoreFilePatterns ?? []
+      ),
     [] as string[]
   )
+  const isIgnoredByCoreListener = createPathMatcher({
+    toMatch: pluginIgnoreContributions,
+  })
 
   const watcher = watch([opts.layout.sourceRoot, ...pluginWatchContributions], {
-    ignored: ['./node_modules', './.*', ...pluginIgnoreContributions],
+    ignored: ['./node_modules', './.*'],
     ignoreInitial: true,
-    onAll(event, file) {
-      for (const p of opts.plugins) {
-        p.dev.onFileWatcherEvent?.(event, file)
-      }
-      restartRunner(file)
-    },
     cwd: process.cwd(), // prevent globbed files and required files from being watched twice
   })
 
+  /**
+   * Core watcher listener
+   */
+  watcher.on('all', (_event, file) => {
+    if (isIgnoredByCoreListener(file)) {
```

Comment:
Is the reason for this more manual approach b/c wtf chokidar basically? 😂

Reply:
Worth a comment I feel, probably at def site, LOC 55.

Reply:
The reason for this is that if we add the ignored files to chokidar directly, then it will ignore files for the core **and** the plugins. In our case, we want to add `prisma.schema` to the watched files, but we only want to ignore `prisma.schema` in the core listener. This is very related to my comment above https://github.com/prisma-labs/graphql-santa/pull/279#discussion_r366957112